### PR TITLE
fix: Individual queries for each period

### PR DESCRIPTION
**Pull request by zubaira**

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramIndicator.java [hunk 1]
```diff
@@ -148,9 +148,7 @@ public void removeIndicatorGroup( ProgramIndicatorGroup group )
      */
     public Boolean hasNonDefaultBoundaries()
     {
-        return this.analyticsPeriodBoundaries.size() != 2 || ( this.analyticsType == AnalyticsType.EVENT && 
-            !this.analyticsPeriodBoundaries.containsAll( defaultEventTypeBoundaries ) ||
-            this.analyticsType == AnalyticsType.ENROLLMENT );
+        return this.analyticsPeriodBoundaries.size() != 2 ||  this.analyticsType == AnalyticsType.EVENT ;
```
Comment (Bekkalizer) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramIndicator.java]: As discussed on slack:
add a useIndividualQueries() method that checks both the hasNonDefaultBoundaries() and a new function hasAnalyticsPeriodVariable()

Comment (zubaira) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramIndicator.java]: fixed

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramIndicator.java [hunk 2]
```diff
@@ -148,10 +152,20 @@ public void removeIndicatorGroup( ProgramIndicatorGroup group )
...
     }
+
+    /**
+     * Checks if expression contains V{analytics_period_end} or V{analytics_period_start}. It will be use in conjunction with hasNonDefaultBoundaries() in order to
+     * split sql queries for each period provided.
+     * @return true if expression has analytics period variables.
+     */
+    public boolean hasAnalyticsVariables()
+    {
+        return ANALYTICS_VARIABLE_PATTERN.matcher( this.expression ).find();
```
Comment (Bekkalizer) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramIndicator.java]: Great, but also include the filter, if the variable is used either in the expression or filter we need this handling.

Comment (zubaira) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramIndicator.java]: fixed

**Pull request merged**